src/app/pages/questionnaire-page/questionnaire-page.component.ts

```diff
@@ -80,17 +83,22 @@ export class QuestionnairePageComponent implements OnInit {
   submitQuestionnaire() {
     this.submitting = true;
 
+    this.genderError = false;
+    this.decimalError = false;
+
     let pageHasErrors = false;
 
     if (!this.gender)
     {
       pageHasErrors = true;
+      this.genderError = true;
     }
 
     // Typecheck to only receive Integers. Without this we get a submit error and exits the page.
     if (this.infantage % 1 != 0)
     {
       pageHasErrors = true;
+      this.decimalError = true;
```

Comment:
This is the tie between the decimalError that we have in the HTML file and the formula to calculate error so the input is equal only to numbers and not equal to zero.